### PR TITLE
Optimize some string interning cases 

### PR DESCRIPTION
**Pull request by Spartan322**
Relies: #169

### CODE REVIEW

File: src/openvic-simulation/dataloader/NodeTools.cpp [hunk 1]
```diff
@@ -159,31 +181,35 @@ node_callback_t NodeTools::expect_colour(callback_t<colour_t> callback) {
...
+												 } else if (!val.is_integer()) {
+													 Logger::warning(
+														 "Fractional part of colour component #", components,
+														 " will be truncated: ", val
+													 );
+												 }
+												 col[components++] = val.to_int64_t();
+												 return true;
+											 }
+										 }))(node);
```
Comment (Hop311) [inline src/openvic-simulation/dataloader/NodeTools.cpp:200]: Formatter sent the lambda function into orbit.

File: src/openvic-simulation/dataloader/NodeTools.cpp [hunk 2]
```diff
@@ -232,10 +258,10 @@ template<typename T, node_callback_t (*expect_func)(callback_t<T>)>
 NodeCallback auto _expect_vec2(Callback<vec2_t<T>> auto callback) {
 	return [callback](ast::NodeCPtr node) -> bool {
 		vec2_t<T> vec;
-		bool ret = expect_dictionary_keys(
-			"x", ONE_EXACTLY, expect_func(assign_variable_callback(vec.x)),
-			"y", ONE_EXACTLY, expect_func(assign_variable_callback(vec.y))
-		)(node);
+		bool ret =
+			expect_dictionary_keys("x", ONE_EXACTLY, expect_func(assign_variable_callback(vec.x)), "y", ONE_EXACTLY, expect_func(assign_variable_callback(vec.y)))(
```
Comment (Hop311) [inline src/openvic-simulation/dataloader/NodeTools.cpp:262]: Formatter combined the key sections.

**Review by Hop311 — APPROVED** [review]

**Comment by Spartan322** [comment]
Best I can do with our current system, this isn't all that optimized without a full compiler.

**Review by Hop311 — APPROVED** [review]
Great work 👍

**Pull request merged**